frontend/projects/abtesting/src/app/features/dashboard/home/components/modal/new-experiment/new-experiment.component.ts

```diff
@@ -57,18 +58,22 @@ export class NewExperimentComponent implements OnInit {
         }
         break;
       case NewExperimentDialogEvents.UPDATE_EXPERIMENT:
+        this.onNoClick();
       case NewExperimentDialogEvents.SAVE_DATA:
         this.newExperimentData = {
           ...this.experimentInfo,
           ...this.newExperimentData,
           ...formData
         };
+        this.openSnackBar();
         this.experimentService.updateExperiment(this.newExperimentData);
-        this.onNoClick();
```

Comment:
If you are removing `this.onNoClick` how will the new experiment or edit experiment dialog be closed after the saving data on the last tab? Is it something that the user have to click on the `CLOSE` from the every stepper?

Reply:
In the last tab it gets auto closed on pressing `create` or `update` button as it is handled in "UPDATE_EXPERIMENT" switch case. On others tab user needs to press `CLOSE` button.